docs/content/noisy_simulation.md

```diff
@@ -0,0 +1,278 @@
+
+## Applicable noise types
+
+The noise models applicable in `Qadence` are classified into three types: digital, analog, and readout error. Digital and analog noise are applicable to `gates`, `blocks`, `circuits`, and `Models`. Readout error is applicable to `sample` and `expectation`. The following is a list of the supported types of noise.
+
+### Digital noise
+
+- BitFlip
+- PhaseFlip
+- Depolarizing
+- PauliChannel
+- AmplitudeDamping
+- PhaseDamping
+- GeneralizedAmplitudeDamping
+
+### Analog noise
+
+- Depolarizing
+- Dephasing
+
+### Readout errors
+
+- Independent
+- Correlated
+
+## NoiseHandler
+
+Noise models can be defined via the `NoiseHandler`. It is a container of several noise instances which require to specify a `protocols` and
+a dictionary of `options` (or lists). The `protocol` field is to be instantiated from `NoiseProtocol`.
+
+```python exec="on" source="material-block" session="noise" result="json"
+from qadence import NoiseHandler
+from qadence.types import NoiseProtocol
+
+digital_noise = NoiseHandler(protocol=NoiseProtocol.DIGITAL.DEPOLARIZING, options={"error_probability": 0.1})
+analog_noise = NoiseHandler(protocol=NoiseProtocol.ANALOG.DEPOLARIZING, options={"noise_probs": 0.1})
+readout_noise = NoiseHandler(protocol=NoiseProtocol.READOUT.INDEPENDENT, options={"error_probability": 0.1, "seed": 0})
+```
+
+
+## Digital noisy simulation
+
+When dealing with programs involving only digital operations, several options are made available from [PyQTorch](https://pasqal-io.github.io/pyqtorch/latest/noise/) via the `NoiseProtocol.DIGITAL`. To use the digital noise option, one needs to state the `protocol` and `error_probability`. All noise types require a single float for the error_probability, while the PauliChannel and GeneralizedAmplitudeDamping require a tuple of error probabilities. One can define noisy digital operations as follows:
+
+```python exec="on" source="material-block" session="noise" result="json"
+from qadence import NoiseProtocol, RX, run
+import torch
+
+noise = NoiseHandler(NoiseProtocol.DIGITAL.BITFLIP, {"error_probability": 0.2})
+op = RX(0, torch.pi, noise = noise)
+
+print(run(op))
+```
+
+It is also possible to set a noise configuration to gates within a composite block or circuit as follows:
+
+```python exec="on" source="material-block" session="noise" result="json"
+from qadence import set_noise, chain, QuantumCircuit
+
+n_qubits = 2
+
+block = chain(RX(i, f"theta_{i}") for i in range(n_qubits))
+
+noise = NoiseHandler(NoiseProtocol.DIGITAL.BITFLIP, {"error_probability": 0.1})
+
+# The function changes the block in place:
+set_noise(block, noise)
+print(run(block))
+
+circuit = QuantumCircuit(2, block)
+noise = NoiseHandler(NoiseProtocol.DIGITAL.BITFLIP, {"error_probability": 0.3})
+set_noise(circuit, noise)
+print(run(circuit))
+```
+
+One can set different noise environments in each gates within the same circuit as follows:
```

Comment:
```suggestion
One can set different noise models for each individual gates within the same circuit as follows:
```

Reply:
Ah, so `set_noise` sets noise globally whereas this example shows local noise setting.